cosmos/config.py

```diff
@@ -322,16 +322,13 @@ def validate_profiles_yml(self) -> None:
     def get_profile_type(self) -> str:
         if isinstance(self.profile_mapping, BaseProfileMapping):
             return str(self.profile_mapping.dbt_profile_type)
+        elif self.profiles_yml_filepath:
+            with open(self.profiles_yml_filepath) as file:
+                profiles = yaml.safe_load(file)
 
-        profile_path = self._get_profile_path()
```

Comment:
By removing this, we are not using the profile caching introduced in #1046 - but this is likely unecessary when checking the profile type. @pankajastro could you share your thoughts on this one, please?